### PR TITLE
TEN-1458 / 24.10 / Adding test case to test a wheel share group

### PR DESCRIPTION
**Pull request by ericbsd**
Also, I sourced everything related to AD

create helper/cmd to put everything related to local command.

Did some code cleanup.

### CODE REVIEW

File: keywords/cmd/smbclient.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        This method verifies if the user can make changes in the share.
+
+        :param share: is the name of the share to access
+        :param user: is the name of the system SMB user
+        :param password: is the password of the SMB user
+
+        :return: True if the user can make changes in the share, otherwise it returns False.
+
+        Example:
+        - SMB.assert_guest_access()
```
Comment (markbainix) [inline keywords/cmd/smbclient.py:19]: incorrect class and method -> `SMBClient.is_user_can_make_changes( x, y, z )`

File: keywords/cmd/smbclient.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        :param share: is the name of the share to access.
+        :param command: is the command to execute in smbclient.
+        :param user: is the name of the system SMB user.
+        :param password: is the password of the SMB user.
+        :param ad: Optional - If true it uses -W AD_WORKGROUP to access the share.
+
+        :return: True if the share can be accessed by the given user, otherwise it returns False.
+
+        Example:
+        - SMB.assert_guest_access()
```
Comment (markbainix) [inline keywords/cmd/smbclient.py:40]: incorrect class and method -> SMBClient.run_smbclient_command( v, w, x, y, z )

File: keywords/ssh/smb.py [hunk 3]
```diff
@@ -89,28 +87,36 @@ def assert_target_command(cls, target: str, share: str, user: str, password: str
...
         Example:
         - SMB.assert_guest_access()
         """
-        response = SSH_Command_Line(f'smbclient //{private_config["IP"]}/{share} -W AD03 -U {user}%{password} -c \'pwd\'', private_config['SMB_ACL_IP'], private_config['SMB_USERNAME'], private_config['SMB_PASSWORD'])
+        usd_ad = f'-W {private_config["AD_WORKGROUP"]} ' if ad else ''
+        command = f'smbclient //{private_config["IP"]}/{share} {usd_ad}-U {user}%{password} -c "pwd"'
+        response = SSH_Command_Line(command, private_config['SMB_ACL_IP'], private_config['SMB_USERNAME'],
+                                    private_config['SMB_PASSWORD'])
+        print('response: ', response.stderr, response.stdout)
         return response.stdout.__contains__(share)
```
Comment (markbainix) [inline keywords/ssh/smb.py:120]: `share in response.stdout`

**Review by markbainix — CHANGES_REQUESTED** [review]

**Review by markbainix — APPROVED** [review]

**Comment by ericbsd** [comment]
image

**Pull request merged**